internal/lib/sandbox/namespaces.go

```diff
@@ -114,11 +115,13 @@ func (s *Sandbox) CreateNamespacesWithFunc(managedNamespaces []NSType, cfg *conf
 				nsPath: namespace.Path(),
 			})
 		case USERNS:
-			s.userns = namespaceIface
-			typesAndPaths = append(typesAndPaths, &ManagedNamespace{
-				nsType: USERNS,
-				nsPath: namespace.Path(),
-			})
+			if idMappings != nil {
```

Comment:
we may want to fail here, otherwise I think we leak the userns (which shouldn't have been created in the first place)

Reply:
the idMappings can be nil, when not using a userns.

Where do we leak the userns?  I'll fix that

Reply:
the way this code reads is:
after we pin the user namespace, verify the idMapping isn't nil. if it is nil, we do not pass the newly pinned user namespace to the caller. If this happens, the caller won't know to clean up the newly created user namespace.

We should fail before the call to `pinns` happens if idMappings is nil but the caller specified USERNS